decorators_library/decorators.py

```diff
@@ -1 +1,85 @@
 # implement your decorators here.
+import time
+# from . import TimeoutError
+class TimeoutError(Exception):
+    def __init__(self, value):
+        self.value = value
+    def __str__(self):
+        return repr(self.value)
+
+# def timeout(func, t):
+#     start_time = time.time()
+#     def new_func(t):
+#         if time.time() - start_time > t:
+#             res = func()
+#             raise TimeoutError("Function call timed out")
+#         return res
+#     return new_func
+
+class timeout(object):
+    def __init__(self, t):
+        self.t = t
+    
+    def __call__(self, func):
+        start_time = time.time()
+        func()
+        total_time = time.time() - start_time
+        if total_time > self.t:
+            raise TimeoutError("Function call timed out")
+        return func
+        
+        
+class memoized(object):
+    def __init__(self, f):
+        self.f = f
+        self.cache = dict()
+        
+    def __call__(self, a, b):
+        if (a,b) in self.cache:
+            return self.cache[(a,b)]
+        res = self.f(a,b)
+        self.cache[(a,b)] = res
+        return res
+        
+'''
+def memoized(f):
```

Comment:
I did a timeout with fibonacci numbers with a memoized version and an un-memoized version. And I must say the performance was sky rocket with memoized